app/server/api/credit_transfer_api.py

```diff
@@ -326,7 +326,7 @@ def post(self, credit_transfer_id):
                     response_list.append({'status': 400, 'message': str(e)})
 
                 else:
-                    db.session.commit()
+                    db.session.flush()
```

Comment:
this doesn't work because a 400 response code doesn't commit.

As I said, I think we should just explicitly commit at the end of a request when we want to, and not do it when we don't.
Otherwise we:
- have a whole bunch of get requests that commit to the db
- have weird inconsistent cases like this one, where we need to commit before make response
- do twice as many flushes as required.